### ISSUE TITLE
smoking rack overrides meat calorie counts

### ISSUE THREAD

**Issue by DavisBroda** [CleverRaven/Cataclysm-DDA#48384]
# Describe the bug

When using a smoking rack, both porkbelly and chunk of meat produce smoked meat as the output. However regular smoked meat is around 400ish calories, while the porkbelly smoked meat has a calorie count of 1300. After initial round of smoking I had 3 porkbelly-smoked-meat, and 42 chunk-of-meat smoked meat. When I picked up the two seperate stacks of smoked meat and inserted them back into the smoking rack, they combined into a single stack. Upon removing them (without turning on the rack), all 45 smoked meat had a calorie count of 1300, rather than either an average of the counts, or splitting into seperate stacks. 

# Steps To Reproduce

1. get porkbelly and smoked meat 
2. insert smoked meat and porkbelly into smoking rack 
3. turn on smoking rack 
4. remove smoked meat from rack (should still be two separate stacks, one with 400ish and one with 1300 calories)
5. put all smoked meat into smoking rack. (it will give you a single interface asking how many, and the max allowable total includes both types of smoked meat)
6. remove smoked meat from rack
7. observe that the entire stack has a single calories value

# Expected behavior

Ideally the two types of smoked meat should remain separate stacks when removed from the smoking rack. Should this not be possible, the resulting combined stack should have a calorie count that is the average of the two inputs (weighted by some combination of item weight and count)

# Screenshots

![high-calorie-smoked-meat](https://user-images.githubusercontent.com/15205060/113655502-f029c800-9667-11eb-95c2-0540bcd26299.PNG)

stack shown was formed from 3 porkbelly smoked meat, and 42 regular smoked meat. note that the entire stack has the 1300 kcal value

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 2004
- Game Version: 0.E-10282-gf22beb8 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions]
]

# Additional context

seems like a similar issue to #40007

**Comment by OliveChukar** [comment]
Tested on 0.E-10347-gf914621 [64-bit].  The issue also happened for me.  I was able to dehydrate human meat mutant human meat normal meat and pork belly together in the smoking rack .  When I removed them put them back in and removed them again they had all turned into the same item type. I tried dropping all the items individually on the ground to see if they would revert to what they originally were when separated and they did not.  Dehydrated fish did not combine with the meat.
